satpy/composites/__init__.py

```diff
@@ -1039,15 +1059,26 @@ def __init__(self, name, filename=None, url=None, known_hash=None, area=None,
 
     @staticmethod
     def _get_cache_filename_and_url(filename, url):
-        if filename is not None:
-            filename = os.path.expanduser(os.path.expandvars(filename))
-        if url is not None:
+        if url:
             url = os.path.expandvars(url)
-            if filename is None:
+            if not filename:
                 filename = os.path.basename(url)
-        if url is None and (filename is None or not os.path.isabs(filename)):
-            raise ValueError("StaticImageCompositor needs a remote 'url' "
-                             "or absolute path to 'filename'.")
+        else:
+            if filename:
+                filename = os.path.expanduser(os.path.expandvars(filename))
+
+                if not os.path.isabs(filename):
+                    data_dir = satpy.config.get('data_dir')
+                    path = os.path.join(data_dir, filename)
+                    if os.path.exists(path):
+                        filename = path
```

Comment:
You could put this chunk of code into a separate method (ex. `def _check_relative_filename(filename):`, could even be a staticmethod I think). That would make codebeat happier too.

Reply:
done.